dali/python/nvidia/dali/pipeline.py

```diff
@@ -969,7 +969,13 @@ def set_outputs(self, *output_data_nodes):
         `*output_data_nodes` : unpacked list of :class:`DataNode` objects
             The outputs of the pipeline
         """
-        self._graph_out = output_data_nodes
+        data_nodes = []
+        for out in output_data_nodes:
+            if isinstance(out, (list, tuple)) and len(out) > 0 and isinstance(out[0], DataNode):
+                data_nodes += list(out)
```

Comment:
This would break if a subgroup was returned from another function that returned a tuple of outputs the first of which is not a data node.
Better way:
```suggestion
            if isinstance(out, (list, tuple)) and any(isinstance(x, DataNode) for x in out):
                data_nodes += list(out)
```